spec/secure-hash.spec.js

```diff
@@ -0,0 +1,44 @@
+"use strict";
+
+describe("secureHash", ()  => {
+    var secureHash, crypto;
+
+    beforeEach(() => {
+        var base64, promiseMock;
+
+        crypto = require("crypto");
+        promiseMock = require("./mock/promise-mock");
+        base64 = require("../lib/base64");
+        secureHash = require("../lib/secure-hash")(base64, crypto, promiseMock);
+    });
+    it("hashes a passed in string", (done) => {
+        secureHash.hashAsync("rRTcBER_EiFUsRa34Hj5Zpok", {
+            algorithm: "sha256",
+            hashLength: 24,
+            iterations: 10000,
+            salt: "pinkFluffyUnicornsDancingOnRainbows"
+        }).then((result) => {
+            expect(result.toString("binary")).toBe("9GnOLZ_xAlfMA4C6DHsjNJJpsShI_TgR");
+        }).then(done, done);
+    });
+    it("hashes a passed in buffer", (done) => {
+        secureHash.hashAsync(new Buffer("rRTcBER_EiFUsRa34Hj5Zpok", "binary"), {
+            algorithm: "sha256",
+            hashLength: 24,
+            iterations: 10000,
+            salt: "pinkFluffyUnicornsDancingOnRainbows"
+        }).then((result) => {
+            expect(result.toString("binary")).toBe("9GnOLZ_xAlfMA4C6DHsjNJJpsShI_TgR");
+        }).then(done, done);
+    });
+    it("hashes without a config being passed in", (done) => {
+        secureHash.hashAsync("rRTcBER_EiFUsRa34Hj5Zpok").then((result) => {
+            expect(result.toString("binary").length).toBe(683);
```

Comment:
What should this result be?  We're not testing the hashed value here, though the defaults should be known and never change.